### ISSUE TITLE
Bad request on multiple Static file locations

### ISSUE THREAD

**Issue by RyanSusana** [perwendel/spark#1005]
Hello friends,

I have been using sparkjava for a while now and am developing a framework ([Elepy](https://github.com/RyanSusana/elepy) using your framework.

The problem that I am running into is that when I have multiple jars with static resources on different Services, I get a Bad Request 400 on all but the first staticfile location. The different Services are hosted on port 80 and 1337 and get instantiated with ignite()

This means that per JVM I can only have one staticfile location? It does recognize that the resource exists, otherwise it would give me a 404 instead.

I looked through the source code and it has something to do with DirectoryTraversalDetection.

**Comment by tipsy** [comment]
Hi @RyanSusana. I'm guessing this is because `StaticFilesFolder` has static variables:

```
private static volatile String local;
private static volatile String external;
```

**Comment by RyanSusana** [comment]
Are there any plans on changing this or must I resort to multiple JVM's?

**Comment by RyanSusana** [comment]
Why would it break backwards compatability?

**Comment by tipsy** [comment]
Any change in current behavior is really a change in backwards compatibility. I don't remember what the current behavior is, but based on reading the code it looks like you can do:

```kotlin
val serviceA = ignite().apply { staticFiles.location("/public") }
val serviceB = ignite()
```

Then do a `GET` for static files to `serviceB` . If this is the case, people will have to add `staticFiles.location("/public") ` to their second service too.

 I'll fire up IntelliJ and check.

**Comment by tipsy** [comment]
Ok, so it didn't work without calling `staticFiles.`, so that's good. Also, I found some strange behavior: 

Works:
```java
Service serviceA = Service.ignite();
serviceA.port(4001);
serviceA.staticFiles.location("/public/pages");
serviceA.init();

serviceA.awaitInitialization();

Service serviceB = Service.ignite();
serviceB.staticFiles.location("/public");
serviceB.port(4002);
serviceB.init();
```

Bad request: 
```java
Service serviceA = Service.ignite();
serviceA.port(4001);
serviceA.staticFiles.location("/public");
serviceA.init();

serviceA.awaitInitialization();

Service serviceB = Service.ignite();
serviceB.staticFiles.location("/public/pages");
serviceB.port(4002);
serviceB.init();
```

This can be fixed as a bug then, I can't imagine anyone really relying on this behavior.

**Comment by RyanSusana** [comment]
Thank you for the help!

**Comment by perwendel** [comment]
Fixed by #1036
